airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py

```diff
@@ -150,28 +150,33 @@ def format_field(field_value, field_type):
         return json.loads(field_value)
 
     if field_type.upper() in date_and_time_snowflake_type_airbyte_type.keys() and field_value:
+
+        if isinstance(field_value, datetime):  # Already formatted date
+            return field_value
+
         try:
             airbyte_format = date_and_time_snowflake_type_airbyte_type[field_type.upper()].get('format', None)
             airbyte_type = date_and_time_snowflake_type_airbyte_type[field_type.upper()].get('airbyte_type', None)
 
             if airbyte_format == 'date-time' and airbyte_type == 'timestamp_with_timezone':
-                if ' ' in field_value:  # Ensure offset is present
+                if ' ' in field_value:  # offset present in response
```

Comment:
This should be the `constant` value, instead of the `literal` comparison. 

```python
TIMESTAMP_OFFSET = " "

# then
...
if TIMESTAMP_OFFSET in field_value:
   ....
   # magic here
```

Same for the rest of the code, that uses this `' '` literal value.